packages/notarization/storage.go

```diff
@@ -94,36 +78,131 @@ func (s *EpochCommitmentStorage) CachedECRecord(ei epoch.Index, computeIfAbsentC
 	return s.ecRecordStorage.Load(ei.Bytes())
 }
 
+func (s *EpochCommitmentStorage) SetFullEpochIndex(ei epoch.Index) error {
```

Comment:
We have here four different indexes and the below functions do the same and just use different key strings. Maybe we can refactor to one general function that receives ei and key string